File: C++/Data Structure/Stack/program1.cpp
#include<iostream>
using namespace std;

struct node
{
    char data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node* PNODE;

class Stack
{
    public:
        PNODE Head;
        int Count;

        Stack();
        void Push(char);       
        void Pop();         
        void Display();
        int CountNode();
};

Stack::Stack()
{
    Head = NULL;
    Count = 0;
}


void Stack::Push(char no)
{
    PNODE newn = NULL;
    newn = new NODE;

    newn->data = no;
    newn->next = NULL;

    newn->next = Head;
    Head = newn;
    Count++;
}

void Stack::Pop()    
{
    char no;   

    if(Count == 0)
    {
        cout<<"Stack is empty"<<endl;
        return;
    }

    no = Head -> data;
    PNODE temp = Head;
    Head = Head->next;
    delete temp;

    Count--;
    cout<<"Removed element is : "<<no<<endl;
}

void Stack::Display()
{
    cout<<"Elements from Stack are : "<<endl;
    PNODE temp = Head;
    while(temp != NULL)
    {
        cout<<temp->data<<" ";
        temp = temp->next;
    }
    cout<<endl;
}

int Stack::CountNode()
{
    return Count;
}

int main()
{
        Stack obj1;

        obj1.Push('A');
        obj1.Push('B');
        obj1.Push('C');
        obj1.Push('D');

        obj1.Display();

        obj1.Pop();
        obj1.Pop();

        obj1.Display();

        obj1.Push('Z');
        obj1.Pop();

        return 0;
}

File: Java/Data Structure/Stack/program1.java
import java.util.*;

class node
{
    public node next;
    public int data;
}

class Stack
{
    public node head;
    public int Count;

    public Stack()
    {
        head = null;
        Count = 0;
    }

    public void Push(int no)
    {
        node newn = new node();

        newn.data = no;
        newn.next = null;

        newn.next = head;
        head = newn;

        Count++;
    }

    public void Pop()
    {
        if(Count == 0)
        {
            return;
        }

        node temp = head;
        head = head.next;
        temp = null;

        Count--;
    }

    public void Display()
    {
        if(Count == 0)
        {
            System.out.println("Stack is empty");
            return;
        }

        node temp = head;

        System.out.println("Elements of Stack are : ");

        while(temp != null)
        {
            System.out.print("|" +temp.data+ "| -> ");
            temp = temp.next;
        }
           System.out.println("NULL");
    }

    public int CountNode()
    {
        return Count;
    }
}

class program1
{
    public static void main(String[] args) 
    {
        Stack obj = new Stack();   

        obj.Push(11);
        obj.Push(21);
        obj.Push(51);
        obj.Push(101);

        obj.Display();

        System.out.println("Number of elements are : "+obj.CountNode());

        obj.Pop();
        obj.Pop();

        obj.Display();

        System.out.println("Number of elements are : "+obj.CountNode());
    }
}
